react/features/shared-video/components/native/AbstractVideoManager.js

```diff
@@ -0,0 +1,288 @@
+
+export const PLAYBACK_STATES = {
+    PLAYING: 'playing',
+    PAUSED: 'pause',
+    STOPPED: 'stop'
+};
+
+/**
+ * Return true if the diffenrece between the two timees is larger than 5.
+ *
+ * @param {number} newTime - The current time.
+ * @param {number} previousTime - The previous time.
+ * @private
+ * @returns {boolean}
+*/
+function shouldSeekToPosition(newTime, previousTime) {
+    return Math.abs(newTime - previousTime) > 5;
+}
+
+/**
+ * The type of the React {@link Component} props of {@link AbstractVideoManager}.
+ */
+export type Props = {
+
+    /**
+     * The current coference
+     */
+    _conference: Object,
+
+    /**
+     * Is the video shared by the local user.
+     *
+     * @private
+     */
+    _isOwner: boolean,
+
+    /**
+     * The shared video owner id
+     */
+    _ownerId: string,
+
+    /**
+     * Updates the shared video status
+     */
+    _setSharedVideoStatus: Function,
+
+    /**
+     * The shared video status
+     */
+     _status: string,
+
+    /**
+     * Seek time in seconds.
+     *
+     */
+    _time: number,
+
+    /**
+     * The video url
+     */
+     _videoUrl: string,
+
+
+    /**
+      * The player's height
+    */
+    height: number,
+
+    /**
+      * The video id
+    */
+    videoId: string,
+
+    /**
+      * The player's width
+    */
+    width: number
+}
+
+/**
+ * Manager of shared video.
+ */
+class AbstractVideoManager extends PureComponent<Props> {
+    throttledFireUpdateSharedVideoEvent: Function;
+
+    /**
+     * Initializes a new instance of AbstractVideoManager.
+     *
+     * @returns {void}
+     */
+    constructor() {
+        super();
+
+        this.throttledFireUpdateSharedVideoEvent = throttle(this.fireUpdateSharedVideoEvent.bind(this), 5000);
+    }
+
+    /**
+     * Implements React Component's componentDidMount.
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        this.processUpdatedProps();
+    }
+
+    /**
+     * Implements React Component's componentDidUpdate.
+     *
+     * @inheritdoc
+     */
+    componentDidUpdate() {
+        this.processUpdatedProps();
+    }
+
+    /**
+     * Implements React Component's componentWillUnmount.
+     *
+     * @inheritdoc
+     */
+    componentWillUnmount() {
+        if (this.dispose) {
+            this.dispose();
+        }
+    }
+
+    /**
+     * Processes new properties.
+     *
+     * @returns {void}
+     */
+    async processUpdatedProps() {
+        const { _status, _time, _isOwner } = this.props;
+
+        if (_isOwner) {
+            return;
+        }
+
+        const playerTime = await this.getTime();
+
+        if (shouldSeekToPosition(_time, playerTime)) {
+            this.seek(_time);
+        }
+
+        if (this.getPlaybackState() !== _status) {
+            if (_status === PLAYBACK_STATES.PLAYING) {
+                this.play();
+            }
+
+            if (_status === PLAYBACK_STATES.PAUSED) {
```

Comment:
Else if?

Reply:
i considered they're already mutually exclusive so it won't matter